lib/src/data/datasource/remote/api_genres_service.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:http/http.dart';
+import '../../../core/util/api_constants.dart';
+import '../../../core/util/data_state.dart';
+import '../../model/genre_model.dart';
+import 'i_api_service.dart';
+
+class ApiGenresService extends IApiService {
+  @override
+  Future<DataState<List<GenreModel>>> fetchDataFromApi() async {
+    try {
+      final Response response = await get(
+        Uri.parse(ApiConstants.baseUrlForGenres),
+        headers: {
+          HttpHeaders.authorizationHeader: ApiConstants.apiBearerToken,
+        },
+      );
+      if (response.statusCode == HttpStatus.ok) {
+        Map<String, dynamic> json = jsonDecode(response.body);
+        List<dynamic> genresFromApi = json['genres'];
+        List<GenreModel> genres = [];
+
+        genres =
+            genresFromApi.map((genre) => GenreModel.fromJson(genre)).toList();
+
+        return DataSuccess<List<GenreModel>>(genres);
```

Comment:
You can wrap all this lines in one, like:
`return DataSuccess<List<GenreModel>>(List<dynamic>.from(json['genres']).map((genre) => GenreModel.fromJson(genre)).toList())`